filebeat/input/event.go

```diff
@@ -58,6 +59,16 @@ func (e *Event) ToMapStr() common.MapStr {
 	return event
 }
 
+// Metadata creates a common.MapStr containing the metadata to
+// be associated with the event.
+func (e *Event) Metadata() common.MapStr {
+	meta := common.MapStr{}
+	if len(e.Pipeline) > 0 {
+		meta["pipeline"] = e.Pipeline
+	}
+	return meta
+}
```

Comment:
output metadata should also work if `meta == nil` =>

```
func (e *Event) Metadata() common.MapStr {
  if e.Pipeline != "" {
    return common.MapStr{
      "pipeline": e.Pipeline,
    }
  }
  return nil

}
```

This way you don't need to create a map if no pipeline is configured.